### PR TITLE
handlers: refactor v1 and v2 handlers

### PR DESCRIPTION
**Pull request by haikuoliu**
### Summary

In Agent `handlers` folder, we have two handler versions: `v1` handlers defined in `handlers` package for Agent introspection endpoint, and `v2` handlers defined in `taskmetadata` package for endpoint that exposes task/container metadata, task/container stats, task IAM role credentials. 

Current handlers structure is quite confusing and makes it hard to add new handlers. So I reorganized the `handlers` folder, my goal is (in descending order of importance):

1. Do not affect handlers functionality, i.e. do not change tests logic, only make changes if it's necessary.
2. Make the structure clear, so that adding new handlers/endpoint is easy.
3. Make `v1` and `v2` handlers' code style consistent, reuse logic if possible.

### Implementation details

1. Create file for each type of handlers (except task and container metadata are in the same file). Move all handlers to `v1` and `v2` packages, define handler path in each handler file. Previously the credentials handler defines the `v1` and `v2` handlers together, so I split it, and make `v2.CredentialsHandler` use the major logic in `v1.CredentialsHandler`.
2. Move `PortResponse` from `v1` package to `v2` package. If we keep `PortResponse` in `v2` package, we will end up with cyclic import which is not supported by Golang.
3. Rename the file for server setup to `${version}_server_setup.go`, `${version}` is for server hostname version, i.e. `v2` server can contain `v2` and `v3` handlers.
4. Move the handler tests that use server setup to `${version}_server_setup_test.go` to avoid cyclic import.
5. Create the `utils` package, move `helper.go` and `types.go` to it.
6. Rename some variables that may be confusing, like `metadata` -> `agent_metadata`, `metadata` -> `task_container_metadata`.
7. Move new response logic to `response.go`.
8. Make some constants/method exported, as we need to use them across packages.
9. Some code style changes, and other minor enhancements.

### Testing

- [x] Builds on Linux (`make release`)
- [x] Builds on Windows (`go build -out amazon-ecs-agent.exe ./agent`)
- [x] Unit tests on Linux (`make test`) pass
- [x] Unit tests on Windows (`go test -timeout=25s ./agent/...`) pass
- [x] Integration tests on Linux (`make run-integ-tests`) pass
- [x] Integration tests on Windows (`.\scripts\run-integ-tests.ps1`) pass
- [x] Functional tests on Linux (`make run-functional-tests`) pass
- [x] Functional tests on Windows (`.\scripts\run-functional-tests.ps1`) pass

New tests cover the changes: yes

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: agent/handlers/license_handler.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+import (
+	"net/http"
+
+	"github.com/aws/amazon-ecs-agent/agent/utils"
+)
+
+var licenseProvider = utils.NewLicenseProvider()
+
+// v1LicenseHandler creates response for '/license' API.
```
Comment (fenxiong) [inline agent/handlers/license_handler.go]: nit: v1LicenseHandler -> licenseHandler

**Comment by richardpen** [comment]
Can each handler be moved to its own package?
```
handlers
├── server_setup.go
├── other_stuff
├── v1
│   ├── task_handlers.go
│   ├── other_handlers.go
├── v2
│   ├── metadata
│   │      |------ metadata.go
│   ├── credentials
│   |        |----credentials.go
```

**Comment by haikuoliu** [comment]
@richardpen @fenxiong 
I figured out a way to resolve the cyclic import problem, I published a new version which defines handlers in `v1` and `v2` packages.

Please review again, thanks😁

File: agent/handlers/v1/license_handler.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	"net/http"
+
+	"github.com/aws/amazon-ecs-agent/agent/utils"
+)
+
+const LicensePath = "/license"
+
+var licenseProvider = utils.NewLicenseProvider()
+
+// LicenseHandler creates response for '/license' API.
```
Comment (fenxiong) [inline agent/handlers/v1/license_handler.go]: I think if we organize handlers via the request path, this one should be at the handlers/ level?

**Comment by haikuoliu** [comment]
@fenxiong Fixed the warnings from golint.

File: agent/handlers/v1/logging_handler.go [hunk 3]
```diff
@@ -11,18 +11,24 @@
 // express or implied. See the License for the specific language governing
 // permissions and limitations under the License.
 
-package handlers
+package v1
```
Comment (fenxiong) [inline agent/handlers/v1/logging_handler.go]: same as above. should be at handlers/ level?

**Comment by fenxiong** [comment]
seems that the tests are failing? https://github.com/haikuoliu/amazon-ecs-agent/commits/branch_v3_handler Make sure tests pass before merging this

**Comment by haikuoliu** [comment]
@fenxiong I have passed the tests and rebased the dev branch.

**Pull request merged**